internal/backend/sessions/sessionworkflows/workflows.go

```diff
@@ -126,7 +127,21 @@ func (ws *workflows) StartWorkflow(ctx context.Context, session sdktypes.Session
 
 func (ws *workflows) sessionWorkflow(wctx workflow.Context, params *sessionWorkflowParams) error {
 	wi := workflow.GetInfo(wctx)
-	sid := params.Data.Session.ID()
+	s := params.Data.Session
+	sid := s.ID()
+
+	// get session time (it was created not long after event dispatch) and then update it with actual event time if found
+	eventTime := s.CreatedAt()
```

Comment:
this is a weird assignment, and probably not accurate 
see my comment above

Reply:
see my comments above
you can't assume timing on session creation and event creation time - they might differ by a lot 

depends on what we want to measure, we need to use its time only